test/host/wasi_nn/wasi_nn.cpp

```diff
@@ -0,0 +1,323 @@
+            });
+
+  return indices;
+}
+} // namespace
+
+TEST(WasiNNTest, OpenVINOBackend) {
+  WasmEdge::Host::WASINN::WasiNNEnvironment Env;
+  WasmEdge::Runtime::Instance::MemoryInstance MemInst(
+      WasmEdge::AST::MemoryType(400));
+  std::array<WasmEdge::ValVariant, 1> Errno = {UINT32_C(0)};
+
+  std::vector<uint8_t> XmlRead, WeightRead, TensorData;
+  std::vector<uint32_t> TensorDim{1, 3, 224, 224};
+  uint32_t BuilderPtr = UINT32_C(0);
+  uint32_t LoadEntryPtr;
+  uint32_t SetInputEntryPtr;
+  uint32_t StorePtr = UINT32_C(65536);
+
+  LoadEntryPtr = BuilderPtr;
+  XmlRead = readBinariesFromDisk("./wasinn_openvino_fixtures/mobilenet.xml");
+  WeightRead = readBinariesFromDisk("./wasinn_openvino_fixtures/mobilenet.bin");
+
+  writeFatPointer(MemInst, StorePtr, XmlRead.size(), BuilderPtr);
+  writeFatPointer(MemInst, StorePtr + XmlRead.size(), WeightRead.size(),
+                  BuilderPtr);
+
+  // wasi-nn load test
+  WasmEdge::Host::WasiNNLoad WasiNNLoad(Env);
+  // meaningless binaries
+  {
+    EXPECT_TRUE(WasiNNLoad.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            LoadEntryPtr, UINT32_C(2), UINT32_C(0), UINT32_C(0), BuilderPtr},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(), static_cast<uint32_t>(ErrNo::Busy));
+  }
+  writeBinaries<uint8_t>(MemInst, XmlRead, StorePtr);
+  writeBinaries<uint8_t>(MemInst, WeightRead, StorePtr + XmlRead.size());
+  StorePtr += (XmlRead.size() + WeightRead.size());
+
+  // wrong builders' length
+  {
+    EXPECT_TRUE(WasiNNLoad.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            LoadEntryPtr, UINT32_C(4), UINT32_C(0), UINT32_C(0), BuilderPtr},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(),
+              static_cast<uint32_t>(ErrNo::InvalidArgument));
+  }
+  // unsupported device. CPU 0, GPU 1, TPU 2
+  {
+    EXPECT_TRUE(WasiNNLoad.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            LoadEntryPtr, UINT32_C(2), UINT32_C(0), UINT32_C(3), BuilderPtr},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(),
+              static_cast<uint32_t>(ErrNo::InvalidArgument));
+  }
+  // successful load
+  {
+    EXPECT_TRUE(WasiNNLoad.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            LoadEntryPtr, UINT32_C(2), UINT32_C(0), UINT32_C(0), BuilderPtr},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(), static_cast<uint32_t>(ErrNo::Success));
+    EXPECT_EQ(*MemInst.getPointer<uint32_t *>(BuilderPtr), 0);
+    BuilderPtr += 4;
+  }
+  // second graph
+  {
+    EXPECT_TRUE(WasiNNLoad.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            LoadEntryPtr, UINT32_C(2), UINT32_C(0), UINT32_C(0), BuilderPtr},
+        Errno));
+    EXPECT_EQ(Errno[0].get<int32_t>(), static_cast<uint32_t>(ErrNo::Success));
+    EXPECT_EQ(*MemInst.getPointer<uint32_t *>(BuilderPtr), 1);
+    BuilderPtr += 4;
+  }
```

Comment:
Please add these tests:
1. The returned graph ID pointer out of bounds.
2. The graph builders out of bounds.
3. The XML and Bin (Ptrs in the graph builder array) out of bounds.